openlibrary/templates/my_books/check_ins/check_in_form.html

```diff
@@ -0,0 +1,64 @@
+$def with()
+
+$code:
+  year = current_year()
+  months = [
+    _('January'),
+    _('February'),
+    _('March'),
+    _('April'),
+    _('May'),
+    _('June'),
+    _('July'),
+    _('August'),
+    _('September'),
+    _('October'),
+    _('November'),
+    _('December'),
+  ]
+
+<div class="check-in">
+  <form class="check-in__form" action="" method="post">
+    <input type="hidden" name="event_type" value="3">
+    <input type="hidden" name="event_id" value="">
+    <input type="hidden" name="edition_key" value="">
+    <div>
+      $_('Add an optional check-in date. Check-in dates are used to track yearly reading goals.')
+    </div>
+    <div class="check-in__inputs">
+      <label class="check-in__label">$_('End Date:')</label>
+      <span>
+        <label class="check-in__year-label">$_('Year:')</label>
+        <select class="check-in__select" name="year">
+          <option value="">$_('Year')</option>
+          <option class="hidden show-if-local-year" value="$(year + 1)">$(year + 1)</option>
+          $for i in range(120):
+            <option value="$(year - i)">$(year - i)</option>
+        </select>
+      </span>
```

Comment:
Oh and another thing: we can avoid having to align these via JS by using this HTML trick. This automatically associated the label without having to specify `for`.

```suggestion
      <label>
        $_('Year:')
        <select class="check-in__select" name="year">
          <option value="">$_('Year')</option>
          <option class="hidden show-if-local-year" value="$(year + 1)">$(year + 1)</option>
          $for i in range(120):
            <option value="$(year - i)">$(year - i)</option>
        </select>
      </label>
```